pinot-transport/src/main/java/com/linkedin/pinot/requestHandler/BrokerRequestHandler.java

```diff
@@ -641,17 +573,17 @@ private BrokerResponse processFederatedBrokerRequest(@Nonnull BrokerRequest orig
    *
    * @param tableName table name.
    * @param responseMap map from server to response.
-   * @param responseSequence response sequence for federated request.
+   * @param isOfflineTable whether the responses are from an OFFLINE table.
    * @param dataTableMap map from server to data table.
    * @param processingExceptions list of processing exceptions.
    */
   private void deserializeServerResponses(@Nonnull String tableName, @Nonnull Map<ServerInstance, ByteBuf> responseMap,
-      int responseSequence, @Nonnull Map<ServerInstance, DataTable> dataTableMap,
+      boolean isOfflineTable, @Nonnull Map<ServerInstance, DataTable> dataTableMap,
       @Nonnull List<ProcessingException> processingExceptions) {
     for (Entry<ServerInstance, ByteBuf> entry : responseMap.entrySet()) {
       ServerInstance serverInstance = entry.getKey();
-      if (responseSequence >= 0) {
-        serverInstance = new ServerInstance(serverInstance.getHostname(), serverInstance.getPort(), responseSequence);
+      if (!isOfflineTable) {
+        serverInstance = new ServerInstance(serverInstance.getHostname(), serverInstance.getPort(), 1);
```

Comment:
I know this was already there in the code before, but can you add some comments on the cases when the same server instance may respond with mulltiple responses? afaik we don't do this. The comment in line 571 somehow implies that this can happen for hybrid tables, so I assume it means it can happen for realtime? Or, is this for locally run integration tests that run both realtime and offline in the same machine? In which case port should be sufficient to differentiate the machiens and you don't need sequence number.